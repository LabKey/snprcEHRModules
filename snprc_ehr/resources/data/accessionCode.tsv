value	description	date_disabled
C	Contract	12-22-2010
G	Dr. Gleiser	9/1/2015
N	Necropsy (Performed by pathology)
P	P&M	9-1/2015
S	Surgical (Biopsy or other tissue(s) subm)