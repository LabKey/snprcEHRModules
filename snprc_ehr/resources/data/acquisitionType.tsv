value	description
0	Unknown
1	Colony-born, Vaginal delivery (at TBRI)
2	Colony-born, Cesarean delivery (at TBRI)
3	Stillbirth/Abortion
4	Trapped by TBRI
5	Received by trade
6	Surgical Removal of dead fetus
7	New Purchase
8	Gift (from outside source)
9	Return of TBRI Loan
10	Outside Loan to TBRI
11	Re-Purchase
12	Admission to TBRI for treatment
13	Admission to TBRI for boarding
14	Purchase, wild-born, known provenance
15	Purchase, wild-born, unknown provenance
16	Purchase, colony-born, other colony
17	Gift, wild-born
18	Gift, colony-born
19	Loan, wild-born
20	Loan, colony-born
21	Returned to TBRI (inadequate model)
22	Colony-born, other colony, by Cesarean Section (dam resides or intending to be returned to SFBR)
23	Colony-born, other colony, by Vaginal delivery (dam resides or intending to be returned to SFBR)
24	Admit to TBRI for quarantine period (tether, surgery...prior to release)
25	NON TxBiomed data acquisition
26	Research related (return to TBRI)
27	Reacquired to complete medical history
28	euthanized in utero, colony management
50	Re-IDed (DBA addition, not physical addition to colony)
97	Parental data only, animal never at TBRI