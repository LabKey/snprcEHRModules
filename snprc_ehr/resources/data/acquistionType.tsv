value	description
0	Unknown
4	Trapped by SFBR
5	Received by trade
7	New Purchase
8	Gift (from outside source)
9	Return of SFBR Loan
10	Outside Loan to SFBR
11	Re-Purchase
12	Admission to SFBR for treatment
13	Admission to SFBR for boarding
14	Purchase, wild-born, known provenance
15	Purchase, wild-born, unknown provenance
16	Purchase, colony-born, other colony
17	Gift, wild-born
18	Gift, colony-born
19	Loan, wild-born
20	Loan, colony-born
21	Returned to SFBR (inadequate model)
24	Admit to SFBR for quarantine period (tether, surgery...prior to release)
25	NON TxBiomed data acquisition
26	Research related (return to SFBR)
27	Reacquired to complete medical history
50	Re-IDed (DBA addition, not physical addition to colony)
97	Parental data only, animal never at SFBR
