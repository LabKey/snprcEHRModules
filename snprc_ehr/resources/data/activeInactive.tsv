value	description
A	Active
I	Inactive