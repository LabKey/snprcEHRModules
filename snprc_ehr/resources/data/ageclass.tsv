label	ageclass	gender	species	min	max
Infant	1		CB	0	0.5
Juvenile	2		CB	0.5	5
Adult	3		CB	5	21
Senior	4		CB	21	
Infant	1		CJ	0	0.25
Juvenile	2		CJ	0.25	1.5
Adult	3		CJ	1.5	8
Senior	4		CJ	8	

Infant	1		MA	0	.25
Adult	3		MA	.25	

Infant	1		MF	0	1
Juvenile	2	U	MF	1	4
Juvenile	2	F	MF	1	4
Juvenile	2	M	MF	1	5
Adult	3	U	MF	4	18
Adult	3	F	MF	4	18
Adult	3	M	MF	5	18
Senior	4		MF	18	

Infant	1		MM	0	1
Juvenile	2	U	MM	1	4
Juvenile	2	F	MM	1	4
Juvenile	2	M	MM	1	5
Adult	3	U	MM	4	18
Adult	3	F	MM	4	18
Adult	3	M	MM	5	18
Senior	4		MM	18	

Infant	1		MN	0	1
Juvenile	2	U	MN	1	4
Juvenile	2	F	MN	1	4
Juvenile	2	M	MN	1	5
Adult	3	U	MN	4	18
Adult	3	F	MN	4	18
Adult	3	M	MN	5	18
Senior	4		MN	18	

Infant	1	U	PC	0	1
Infant	1	M	PC	0	1
Infant	1	F	PC	0	1
Juvenile	2	U	PC	1	4
Juvenile	2	F	PC	1	4
Juvenile	2	M	PC	1	5
Adult	3	U	PC	4	18
Adult	3	F	PC	4	18
Adult	3	M	PC	5	18
Senior	4		PC	18	

Infant	1		PT	0	5
Juvenile	2		PT	5	7
Adult	3		PT	7	34
Senior	4		PT	34	

Infant	1		SB	0	1
Juvenile	2		SB	1	3
Adult	3		SB	3	

Infant	1		SM	0	0.25
Juvenile	2		SM	0.25	1.5
Adult	3		SM	1.5	8
Senior	4		SM	8	