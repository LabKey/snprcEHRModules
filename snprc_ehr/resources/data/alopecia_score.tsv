value	sort_order	description
0	1	no alopecia
1	2	>0 - 10% of the body affected
2	3	>10 - 25% of the body affected
3	4	>25 - 50% of the body affected
4	5	>50 - 75% of the body affected
5	6	>75% of the body affected