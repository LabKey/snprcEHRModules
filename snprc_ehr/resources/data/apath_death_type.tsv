value	description
N	Found dead/Natural
R	Research-related/Other than euthanasia
S	Sacrifice/Research-related
E	Euthanasia/Humane
U	Unknown