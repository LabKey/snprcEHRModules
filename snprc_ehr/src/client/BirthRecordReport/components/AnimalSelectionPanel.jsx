import React from 'react'
import Select from 'react-select'

export default class AnimalSelectionPanel extends React.Component {
    handleChange = option => {
        this.props.handleChange(option)
    }
render() {
        return (
          <>
            <div className="wizard-panel__rows">
              <div className="wizard-panel__row">
                <div className="wizard-panel__col">
                  <label className="field-label-align-close">Animal</label>
                  <Select
                    autoFocus
                    className="shared-dropdown"
                    classNamePrefix="shared-select"
                    id="animal-select"
                    isClearable
                    isLoading={ !this.props.animalList }
                    onChange={ this.handleChange }
                    options={ this.props.animalList }
                    placeholder="Select Animal"
                    value={ this.props.selectedAnimal || null }
                  />
                </div>
              </div>
            </div>
          </>
        )
    }
}
