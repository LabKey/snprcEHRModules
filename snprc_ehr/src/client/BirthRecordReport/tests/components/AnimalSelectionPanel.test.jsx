import React from 'react'
import { shallow } from 'enzyme'
import AnimalSelectionPanel from '../../components/AnimalSelectionPanel'
import data from '../fixtures/testData'

test('Should render the AccountPanel', () => {
    const { animalList } = data
    const wrapper = shallow(
      <AnimalSelectionPanel
        animalList={ animalList }
        handleChange={ () => { } }
        selectedAnimal={ undefined }
      />
    )
    expect(wrapper).toMatchSnapshot()
})
