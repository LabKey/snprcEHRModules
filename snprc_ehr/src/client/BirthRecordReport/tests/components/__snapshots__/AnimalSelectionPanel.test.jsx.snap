// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the AnimalSelectionPanel 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
  >
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label-align-close"
        >
          Animal
        </label>
        <ForwardRef
          autoFocus={true}
          className="shared-dropdown"
          classNamePrefix="shared-select"
          id="animal-select"
          isClearable={true}
          isLoading={false}
          onChange={[Function]}
          options={
            [
              {
                "AcqDate": "2020-06-11 15:00:00.000",
                "AcquisitionType": 1,
                "AnimalAccount": "0",
                "BirthCode": 0,
                "BirthDate": "2020-06-11 15:00:00.000",
                "Cage": 2,
                "Colony": 214,
                "Diet": "2",
                "IACUC": "0",
                "Id": "278",
                "OwnerInstitution": 1,
                "ResponsibleInstitution": 1,
                "dam": null,
                "gender": "F",
                "label": "Id: 278 - Female - Birth Date: 06/11/2020 3:00 PM ",
                "pedigree": null,
                "room": 4201,
                "sire": null,
                "species": "PCA",
                "value": "278",
              },
            ]
          }
          placeholder="Select Animal"
          value={null}
        />
      </div>
    </div>
  </div>
</Fragment>
`;
